lib/elix/responders/lists.ex

```diff
@@ -81,20 +65,31 @@ defmodule Elix.Responders.Lists do
   delete <item> from <name> - Deletes an item from a list by name or number
   """
   respond ~r/delete (.+) from (.+)/i, %Message{matches: %{1 => item_id, 2 => list_id}} = msg do
-    list_name = if Regex.match?(~r/\A\d+\Z/, list_id) do
+    list_name = parse_list_identifier(list_id)
+    item_name = parse_item_identifier(item_id, list_name)
+
+    Lists.delete_item(list_name, item_name)
+    reply(msg, render_list(list_name))
+  end
+
+  defp parse_list_identifier(list_id) do
+    if numeric_string?(list_id) do
```

Comment:
Having re-read that I agree. Best it would get us is `is_bitstring` which isn't super valuable. I also tried to think of a way to do destructuring with regex, but it wasn't coming together. Here is a wild ride I think would work, but is also a terrible idea:

```elixir
defp parse_list_identifier(list_id) do
  parse_list_identifier(list_id, "")
end

defp parse_list_identifier([], list_id) do
  list_id
  |> String.to_integer
  |> Lists.get_name
end

defp parse_list_identifier([h|t], list_id) when is_numericstring(h) do
  parse_list_identifier(t, list_id <> h)
end

defp parse_list_identifier(remainder, list_id) do
  list_id <> remainder
end


defmacro is_numericstring(char) do
  quote do
    unquote(char) == "0" ||
      unquote(char) == "1" ||
      unquote(char) == "2" ||
      unquote(char) == "3" ||
      unquote(char) == "4" ||
      unquote(char) == "5" ||
      unquote(char) == "6" ||
      unquote(char) == "7" ||
      unquote(char) == "8" ||
      unquote(char) == "9"
  end
end
```